python-checks/src/test/resources/checks/pandasToDatetimeFormatCheck.py

```diff
@@ -0,0 +1,155 @@
+import pandas as pd
+
+
+def yearfirst_true_noncompliant():
+    pd.to_datetime("02-03-2022", yearfirst=True) # Noncompliant {{Remove this `yearfirst=True` parameter or make sure the provided date(s) can be parsed accordingly.}}
+#                  ^^^^^^^^^^^^> ^^^^^^^^^^^^^^
+    pd.to_datetime(
+        "02/03/2022",
+    #   ^^^^^^^^^^^^>  {{Invalid date.}}
+        yearfirst=True # Noncompliant
+    ) # ^^^^^^^^^^^^^^
+
+    pd.to_datetime("02 03 2022", yearfirst=True) # Noncompliant
+    pd.to_datetime("02.03.2022", yearfirst=True) # Noncompliant
+    pd.to_datetime("02;03;2022", yearfirst=True) # Noncompliant
+    pd.to_datetime("02_03_2022", yearfirst=True) # Noncompliant
+
+
+def list_argument_noncompliant():
+    pd.to_datetime(["02-03-2022"], yearfirst=True) # Noncompliant
+    pd.to_datetime(
+        ["02-03-2022", "03-03-2022"],
+    #   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^> {{This contains invalid date(s).}}
```

Comment:
I think it would be better to assert that there is only 1 secondary location, instead of the 2 the user might expect

Reply:
There are actually 2 secondary locations (the assertion for the second one is on the line below with a `@-1` specifier). I figure if the list is long, showing an example of the invalid literal is still interesting.
I added some assertions on the order of secondary locations to make things more explicit.